envoy/api/v2/route/route.proto

```diff
@@ -208,6 +208,28 @@ message WeightedCluster {
     // cluster with metadata matching that set in metadata_match will be
     // considered. The filter name should be specified as *envoy.lb*.
     Metadata metadata_match = 3;
+
+    // [#not-implemented-hide:]
+    // Specifies a list of headers to be added to requests when this cluster is selected
+    // through the enclosing :ref:`envoy_api_msg_route.RouteAction`.
+    // Headers specified at this level are applied before headers from the enclosing
+    // :ref:`envoy_api_msg_route.RouteAction`,
+    // :ref:`envoy_api_msg_route.VirtualHost`, and
+    // :ref:`envoy_api_msg_route.RouteConfiguration`. For more information, including details on
+    // header value syntax, see the documentation on :ref:`custom request headers
+    // <config_http_conn_man_headers_custom_request_headers>`.
+    repeated HeaderValueOption request_headers_to_add = 4;
+
+    // [#not-implemented-hide:]
+    // Specifies a list of headers to be added to responses when this cluster is selected
+    // through the enclosing :ref:`envoy_api_msg_route.RouteAction`.
+    // Headers specified at this level are applied before headers from the enclosing
+    // :ref:`envoy_api_msg_route.RouteAction`,
+    // :ref:`envoy_api_msg_route.VirtualHost`, and
+    // :ref:`envoy_api_msg_route.RouteConfiguration`. For more information, including details on
+    // header value syntax, see the documentation on :ref:`custom request headers
+    // <config_http_conn_man_headers_custom_request_headers>`.
+    repeated HeaderValueOption response_headers_to_add = 5;
```

Comment:
Should we go ahead an put `response_headers_to_remove` in here?

Reply:
+1 might as well be consistent

Reply:
Added `response_headers_to_remove`

Now it feels like it should be a separate proto, there is plenty of repetition in this file.
```proto
message HeaderAction {
   repeated HeaderValueOption request_headers_to_add = 1;
   repeated HeaderValueOption response_headers_to_add = 2;
   repeated string response_headers_to_remove = 3;
}
```

Reply:
Unfortunately, we can't refactor, as the other locations that this occurs in are frozen.